tests/test_snapshot_download.py

```diff
@@ -0,0 +1,19 @@
+import os
+import unittest
+
+from huggingface_hub.snapshot_download import snapshot_download
+
+
+SNAPSHOT_MODEL_ID = "lysandre/pair-classification-roberta-mnli"
+SNAPSHOT_MODEL_ID_REVISION_MAIN = "084226a48571f3e227230bdbdb7cad43b98efc44"
+# Commit at the top of main
+# todo(replace with a smaller model as this test downloads large files.)
+
+
+class SnapshotDownloadTests(unittest.TestCase):
+    def test_download_model(self):
+        storage_folder = snapshot_download(SNAPSHOT_MODEL_ID, revision="main")
```

Comment:
Maybe specify `cache_dir=` here to use a temporary directory ?
Otherwise if you've already downloaded the repo, it would reload it from the cache without properly testing the download part.